pageserver/src/tenant/storage_layer/delta_layer.rs

```diff
@@ -1442,6 +1442,19 @@ impl DeltaLayerInner {
     }
 
     pub fn iter<'a>(&'a self, ctx: &'a RequestContext) -> DeltaLayerIterator<'a> {
```

Comment:
followup: WDYT about removing the `iter()` with defaults?  I think folks using this method will not guess that it has an 8MB footprint, it might be safer to force callers to always explicitly say how much memory they want to use